e2e/mock/peer.go

```diff
@@ -30,15 +31,15 @@ func (p *Peer) DeliverFiltered(srv peer.Deliver_DeliverFilteredServer) error {
 		panic("expect no recv error")
 	}
 	srv.Send(&peer.DeliverResponse{})
-
-	for range p.TxC {
-		p.txCnt++
-		if p.txCnt%p.BlkSize == 0 {
-			srv.Send(&peer.DeliverResponse{Type: &peer.DeliverResponse_FilteredBlock{
-				FilteredBlock: &peer.FilteredBlock{FilteredTransactions: make([]*peer.FilteredTransaction, p.BlkSize)}}})
+	if !p.suspend {
```

Comment:
this would not work:
1. there's race around access to `suspend`
2. once this for-loop is started, `Suspend` would not be effective.